hibernate-core/src/main/java/org/hibernate/type/descriptor/java/BooleanTypeDescriptor.java

```diff
@@ -159,7 +158,7 @@ public int getDefaultSqlScale() {
 
 	@Override
 	public String getCheckCondition(String columnName, SqlTypeDescriptor sqlTypeDescriptor, Dialect dialect) {
-		return sqlTypeDescriptor instanceof BitTypeDescriptor && !dialect.supportsBitType()
+		return dialect.toBooleanValueString(true).equals("1") && !dialect.supportsBitType()
 				? columnName + " in (0,1)"
 				: null;
```

Comment:
I'm thinking we should now move this logic to `Dialect`. Add a new `getBooleanCheckConstraint()` method or something like that.

Reply:
I had to change the implementation to respect the `SqlTypeDescriptor` again because otherwise types like e.g. `YesNoType` or `TrueFalseType` wouldn't work anymore.

Reply:
I'm not objecting to this change. I'm just saying that instead of having this fragment of code

```
dialect.toBooleanValueString(true).equals("1") && !dialect.supportsBitType()
				? columnName + " in (0,1)"
				: null;
```

in `BooleanTypeDescriptor`, which is not supposed to know about rendering SQL, we move that fragment of code to a new method of `Dialect`, and dispatch from `BooleanTypeDescriptor. getCheckCondition()` back to `Dialect.getBooleanCheckConstraint()`.

It's a cleaner separation of concerns, and potentially more flexible.

Reply:
Since the check constraint depends on the `true`/`false` encodings and `org.hibernate.type.descriptor.java.JavaTypeDescriptor#getCheckCondition` is also implemented by `EnumJavaTypeDescriptor` I'd rather leave this as it is for now. I agree that `org.hibernate.type.descriptor.java.JavaTypeDescriptor#getCheckCondition` is not necessarily the "right place" for generating a check constraint. Maybe moving this to `org.hibernate.type.BasicType` is more appropriate? Anyway, I think this should be done in a separate PR unless you insist on doing this here already.

Reply:
No that's fine. I don't think you've understood what I'm saying, so I'll just do it myself later.